adapters/adapterstest/test_json.go

```diff
@@ -190,8 +190,15 @@ func assertMakeRequestsOutput(t *testing.T, filename string, actual []*adapters.
 	if len(expected) != len(actual) {
 		t.Fatalf("%s: MakeRequests had wrong request count. Expected %d, got %d", filename, len(expected), len(actual))
 	}
-	for i := 0; i < len(actual); i++ {
-		diffHttpRequests(t, fmt.Sprintf("%s: httpRequest[%d]", filename, i), actual[i], &(expected[i].Request))
+
+	for i := 0; i < len(expected); i++ {
+		var err error
+		for j := 0; j < len(actual); j++ {
+			if err = diffHttpRequests(fmt.Sprintf("%s: httpRequest[%d]", filename, i), actual[j], &(expected[i].Request)); err == nil {
+				break
+			}
```

Comment:
Not sure about this approach. Please correct me if I'm wrong. For example it may not work correctly in case `expected` has 2 same exact requests (IDK if this is the case) and actual has also 2 requests, one is correct and another one is empty(or different). Bacause you start iteration over actual from the beginning - such case will pass. 
I feel like you need to remove matched actual request from collection after it matched. 
I was not able to create a unit tests for this, becuse in most cases we only have 1 expected request. I can try to do this when 33across changes will be merged. 
This test needs a test :)

Reply:
I think Veronika is correct in bringing up this example that wouldn't work. I'm also not sure if this behavior of 2 same requests is possible, but the solution to remove matched request from list makes sense to me!

Another idea could be to maybe convert ```expected``` and ```actual``` lists to be maps, which could make the comparison of these objects faster in general and solve the issue Veronika brought up.

Reply:
@AlexBVolcy Maps instead of a nested for loop was my first choice but I tried `map[string]*RequestData` but couldn't get the keys to match. Maybe I need to try again because I definitely like the hash table approach better.

Reply:
> Not sure about this approach. Please correct me if I'm wrong. For example it may not work correctly in case `expected` has 2 same exact requests (IDK if this is the case) and actual has also 2 requests, first one is correct and second one is empty(or different/incorrect). 

You are right, this approach was implemented under the assumption that the `expected` entries at least differ in the `root.imp[i].id` fields but a corner case should be considered where this is not true. I'll give the remove matched `actual` request approach a try.